src/screens/MainScreen.tsx

```diff
@@ -1,37 +1,102 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { StyleSheet } from 'react-native';
 import colors from '@styles/colors';
 import { GestureHandlerRootView } from 'react-native-gesture-handler';
 import PagerView from 'react-native-pager-view';
 import { CurrentScreen } from '@screens/CurrentScreen';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { LOCATIONS } from '@storage/constants';
+import { ILocation } from '@storage/types';
+import * as Location from 'expo-location';
+
+const apiKey = 'f467ce1b7a6266168a069f38c99d7029';
 
 export const MainScreen = () => {
     const [currentIndex, setCurrentIndex] = useState(0);
 
+    const [locations, setLocations] = useState([]);
+
+    const getDataFromStorage = async () => {
+        const locationsRaw = await AsyncStorage.getItem(LOCATIONS);
+        console.log(locationsRaw);
+        const loc = (JSON.parse(locationsRaw) as Array<ILocation>) || [];
+        setLocations(loc);
+        return loc;
+    };
+
+    useEffect(() => {
+        (async () => {
+            updateStorage([
+                {
+                    location: 'Омск',
+                    locationId: '231231',
+                    timeZone: 'Asia/Omsk',
+                    latitude: -33.32,
+                    longitude: -122.23,
+                },
+            ]);
+
+            const storagelocations = await getDataFromStorage();
+            const coords = await getCurrentLocation();
+            if (coords === null) return;
+            const weather = await getWeather(coords);
+
+            if (weather === null) return;
+            if (locations.some(val => val.id === weather.id)) {
```

Comment:
Change to !locations.some... , because we want to add new location if it doesn't exist in array